server/auth-strategies/ldap.js

```diff
@@ -28,26 +29,96 @@ function enableLdap(config) {
           bindCredentials:
             config.get('ldapPassword') || config.get('ldapPassword_d'),
           searchFilter: config.get('ldapSearchFilter'),
+          groupSearchBase: config.get('ldapBaseDN'),
+          groupSearchFilter: '(cn={{dn}})',
         },
       },
       async function passportLdapStrategyHandler(req, profile, done) {
         try {
           const { models } = req;
-          const mail = profile.mail.toLowerCase();
-          const user = await models.users.findOneByEmail(mail);
-          if (!user) {
+
+          const uid = profile.uid.toLowerCase();
```

Comment:
This line need change to `const uid = (profile.uid || profile.sAMAccountName).toLowerCase();`, because uid properties in ActiveDirectory is `sAMAccountName`.
When I change this line, the test is successful.